### PR TITLE
[PLAT-477]containerization airqo website

### PR DESCRIPTION
**Pull request by danielogen**
#### Summary of Changes (What does this PR do?)
- This PR adds docker containerization to the current airqo website
#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
To test this PR, make sure you have [docker](https://www.docker.com/) installed. Checkout this branch locally and `cd` to `AirQo-frontend/web`.  Get the `.env` from the PR author. Also add the database dump (e.g. airqo_db.sql) in `db_data` folder.

Then run the following steps:
Step 1: Build docker images 
```
docker-compose build
``` 
The command above will build three docker images i.e. `PHP apache server`, `MySQL DB server` and `Adminer` - a web frontend for managing MySQL DB.

Step 2: Start the containers and create a network to connect them together-
```
docker-compose up
```
Once all the containers are fully configured, go to your browser and point it at `http://localhost:8000`. Admin panel is at `http://localhost:8000/admin`.
if you wish to browse the DB, goto `http://localhost:8001`. The credentials are provided in the `.env` file.

Step 3: Clean up
 You can hit `ctrl+c` or `cmd+c`, then, run `docker-compose down`.

**NB: Ensure that the following ports are not be utilize by other services: `8000`, `8001`, `8806`.**

#### What are the relevant tickets?
- [PLAT-477]

#### Screenshots
Screenshot 2021-04-08 at 10 07 48
Screenshot 2021-04-08 at 10 07 25
Screenshot 2021-04-08 at 10 07 13

[PLAT-477]: https://airqoteam.atlassian.net/browse/PLAT-477

### CODE REVIEW

**Review by NoahNsimbe — APPROVED** [review]
LGTM

**Comment by NoahNsimbe** [comment]
@danielogen, system folder should not be tracked.
The codebase looks outdated. Compare the application Home page and careers page to [Airqo website](https://www.airqo.net/)

**Comment by danielogen** [comment]
Hi @NoahNsimbe, thanks for the review. I will remove the system folder but remember without it, the website won't run. 
For the case of outdated codebase, I think that's beyond the scope of this PR. I'm basically using the latest staging code. cc: @Baalmart

**Review by Baalmart — COMMENTED** [review]
Thanks @danielogen , could we a add a quick README with the details of how to get things up and running. Things like the link to get the appropriate `systems folder`, etc....

**Comment by danielogen** [comment]
> Thanks @danielogen , could we a add a quick README with the details of how to get things up and running. Things like the link to get the appropriate `systems folder`, etc....

Thanks @Baalmart. This is done!

File: web/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+### Local Setup Using Docker 
+The AirQo website can be configured locally for development and testing purposes.
+#### **Requirements**
+- [Docker](https://www.docker.com/)
+- Download and extract [CodeIgniter](https://codeigniter.com/) System folder at the root of this directory. Link: https://api.github.com/repos/bcit-ci/CodeIgniter/zipball/refs/tags/3.1.11
+- Open ports: 8000, 8001, 88036
+
+#### **Setup**
+
+Checkout this branch locally and `cd` to `AirQo-frontend/web`.  Get the `.env` from the PR author. Also add the database dump (e.g. airqo_db.sql) in `db_data` folder.
```
Comment (ja-odur) [inline web/README.md:13]: `Get the .env from the PR author` lol... We will be coming to you in the future 😃

**Review by ja-odur — COMMENTED** [review]
I tried running the setup, I could assess the adminer on `http://localhost:8001/` but could not access the app on `http://localhost:8000/`. I followed instructions in the README.md. Probably I missed something

File: web/README.md (hunk 1 above)
Comment (danielogen) [reply web/README.md:13]: lol.. .env was shared in our multi-DM

Comment (ja-odur) [reply web/README.md]: @danielogen I managed to get the .env and the sql file. I cant access the app when I run docker-compose up

Comment (NoahNsimbe) [reply web/README.md]: @ja-odur, do you have the magical system folder????

Comment (ja-odur) [reply web/README.md]: @NoahNsimbe I downloaded that too. added it to the root folder (web)

**Review by sserurich — APPROVED** [review]
could access the  website following the instructions in the readme.. just that the nodes on the map aren't displayed but looking at the screenshot you shared, also it didn't have nodes.. i

**Comment by sserurich** [comment]
> could access the website following the instructions in the readme.. just that the nodes on the map aren't displayed but looking at the screenshot you shared, also it didn't have nodes.. 

@danielogen , is it expected behavior for this PR.

**Review by ja-odur — COMMENTED** [review]
@danielogen I am getting this error, not sure if it's unique to my docker version

image

**Comment by danielogen** [comment]
> @danielogen I am getting this error, not sure if it's unique to my docker version
> 
> image

oh I see, I think you are missing `-`. Try using `docker-compose`

**Comment by ja-odur** [comment]
> > @danielogen I am getting this error, not sure if it's unique to my docker version
> > image
> 
> oh I see, I think you are missing `-`. Try using `docker-compose`
Jesus Christ!

**Review by ja-odur — COMMENTED** [review]
@danielogen  same error. This new docker is kinder strict
image

**Comment by danielogen** [comment]
> @danielogen same error. This new docker is kinder strict
> image

Looking at the warnings above, seems you are missing `.env` as well?? Anyway, let me continue to explore. Could you share the version of your docker? Mine is `Docker version 20.10.2, build 2291f61`. Thanks!

**Comment by ja-odur** [comment]
> > @danielogen same error. This new docker is kinder strict
> > image
> 
> Looking at the warnings above, seems you are missing `.env` as well?? Anyway, let me continue to explore. Could you share the version of your docker? Mine is `Docker version 20.10.2, build 2291f61`. Thanks!

The versions I am using `docker 20.10.5` and `compose 1.28.5`

**Review by ja-odur — APPROVED** [review]
Works fine, the issue was the missing `.env` file.

**Comment by danielogen** [comment]
> > could access the website following the instructions in the readme.. just that the nodes on the map aren't displayed but looking at the screenshot you shared, also it didn't have nodes..
> 
> @danielogen , is it expected behavior for this PR.

Yes @sserurich , this expected. Once we have the latest version from @buzentech , It should be okay!

**Pull request merged**